### ISSUE TITLE
Implement Layer Normalization (`LayerNorm`)

### ISSUE THREAD

**Issue by FL03** [pzzld-org/concision#62]
Layer normalization directly estimates the normalization statistics from the summed inputs to the neurons in a _**hidden**_ layer.

# Model

## Parameters

The implementation can leverage the [ParamsBase](https://docs.rs/concision/latest/concision/linear/params/struct.ParamsBase.html) created for linear models.

- Bias (optional)
- Weight

<details>
<summary>References</summary>
<p>

- [Layer Normalization](https://arxiv.org/abs/1607.06450)
- [Understanding and Improving Layer Normalization](https://arxiv.org/pdf/1911.07013)
- [torch.nn.LayerNorm](https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html#torch.nn.LayerNorm)

</p>
</details>
